subsystem/elevator.py

```diff
@@ -0,0 +1,204 @@
+    turn_sensor = rev.CANSparkMax(config.elevator_main_rotation_motor_id,
+                                  rev.MotorType.kBrushless).getAnalog()  # senses the rotation of the elevator
+    brake_enabled: bool = False
+    pose = Pose3d(0, 0, constants.pivot_point_height, 0, 0, 0)
+    elevator_length: float # the length of the elevator in meters
+    rotation_overide: bool = False  # if true, the rotation limits will be ignored
+    extension_overide: bool = False  # if true, the extension limits will be ignored
+    brake_overide: bool = False  # if true, the brake will not be enabled or disabled
+    intake_up: bool = True
+
+    def enable_brake(self):
+        if self.brake_overide:
+            return
+        else:
+            self.brake_enabled = True
+            self.brake.set(True)
+
+    def disable_brake(self):
+        if self.brake_overide:
+            return
+        else:
+            self.brake_enabled = False
+            self.brake.set(False)
+
+    def get_pose(self):
+        # returns the pose of the arm
+        return self.pose
+
+    def set_pose(self, pose: Pose3d):
+        self.pose = pose
+
+    def is_at_position(self, position: Pose3d):
+        if (self.get_pose() == position):
+            return True
+        else:
+            return False
+
+    def init(self):  # initializing motors
+        self.motor_extend.init()
+        self.secondary_rotation_motor.init()
+        self.main_rotation_motor.init()
+        self.secondary_rotation_motor.follow(
+            config.elevator_main_rotation_motor_id, True)
+        self.enable_brake()
+
+    def stop(self):
+        self.motor_extend.set_raw_output(0)
+        self.main_rotation_motor.set_raw_output(0)
+
+    def hard_stop(self):
+        self.stop()
+        self.enable_brake()
+
+    def boundary_box(self, angle: float): #TESTED AND WORKING
+
+        if self.extension_overide:
+            return 1
+        else:
+            angle = abs(angle)
+
+            def get_length_ratio(angle, adjacent):
+                # Gets the length of the angle using cosine
+                max_length = adjacent / math.cos(angle)
+                # if the length of the angle is greater than the max_elevator_height, return one
+                # else give a percentage of the height of the current line and the max_elevator_height
+                # protects against any irregular numbers heigher than the max elevator height by always
+                # defaulting to 1
+                if abs(max_length) > constants.max_elevator_height:
+                    return 1
+                else:
+                    res = abs(max_length / constants.max_elevator_height)
+                    return res  # where is the hype at
+
+            # Finds the length of each the dimensions for the boundry box using the constants
+            top_vertical_boundry_from_pivot = constants.vertical_boundary - constants.pivot_point_height - constants.top_boundry_buffer_gap
```

Comment:
`top_vertical_boundry_from_pivot` and the other values derived from constants can be calculated in init or even in the constants file which would make this function easier to read.